### PR TITLE
fix: UTM mobile

### PR DESCRIPTION
**Pull request by teolemon**
### Screenshot
fix: UTM mobile

### CODE REVIEW

File: templates/web/common/site_layout.tt.html [hunk 1]
```diff
@@ -51,6 +51,7 @@
 									<li><a href="[% edq(lang('menu_contribute_link')) %]">[% lang('menu_contribute') %]</a></li>
 									<li class="divider"></li>
 									<li><label>[% lang("add_products") %]</label></li>
+                <!-- For reference: get_the_app_link_off" = /open-food-facts-mobile-app" -->
 									<li><a href="[% edq(lang_flavor('get_the_app_link')) %]?utm_source=off&utf_medium=web&utm_campaign=pro_platform_install_the_app_to_add_products_[% language %]">[% lang('install_the_app_to_add_products') %]</a></li>
```
Comment (hangy) [inline templates/web/common/site_layout.tt.html:55]:
```suggestion
									<li><a href="[% edq(lang_flavor('get_the_app_link')) %]&utm_source=off&utf_medium=web&utm_campaign=pro_platform_install_the_app_to_add_products_[% language %]">[% lang('install_the_app_to_add_products') %]</a></li>
```

File: templates/web/common/site_layout.tt.html [hunk 2]
```diff
@@ -242,13 +244,15 @@
 				<!-- full width banner on mobile -->
 				
 				[% IF mobile %]
-					<a href="[% mobile.link %]?utm_source=off&utf_medium=web&utm_campaign=mobile-app-top-banner-[% language %]" id="mobile-app-top-banner" class="button large expand success">
+        <!-- For reference: mobile.link = https://play.google.com/store/apps/details?id=org.openfoodfacts.scanner -->
+					<a href="[% mobile.link %]&utm_source=off&utf_medium=web&utm_campaign=mobile-app-top-banner-[% language %]" id="mobile-app-top-banner" class="button large expand success">
```
[the remark this answers is not in the document]
Comment (stephanegigandet) [reply templates/web/common/site_layout.tt.html:248]: typo: utf_medium

**Review by stephanegigandet — CHANGES_REQUESTED** [review]
Add utm_medium and utm_source in Config_[off|obf...].pm and then only add ?utm_campaign

File: templates/web/common/site_layout.tt.html (hunk 1 above)
Comment (teolemon) [reply templates/web/common/site_layout.tt.html:55]: There's no question mark in the root URL, I believe @hangy

File: templates/web/common/site_layout.tt.html (hunk 2 above)
Comment (teolemon) [inline templates/web/common/site_layout.tt.html:248]:
```suggestion
					<a href="[% mobile.link %]&utm_source=off&utm_medium=web&utm_campaign=mobile-app-top-banner-[% language %]" id="mobile-app-top-banner" class="button large expand success">
```

File: lib/ProductOpener/Config_obf.pm [hunk 3]
```diff
@@ -177,9 +177,9 @@ $flavor = 'obf';
 	site_name => "Open Beauty Facts",
 	product_type => "beauty",
 	og_image_url => "https://world.openbeautyfacts.org/images/misc/openbeautyfacts-logo-en.png",
-	android_apk_app_link => "https://world.openbeautyfacts.org/images/apps/obf.apk",
-	android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner",
-	ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380",
+	android_apk_app_link => "https://world.openbeautyfacts.org/images/apps/obf.apk?utm_source=off&utf_medium=web",
+	android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner&utm_source=off&utf_medium=web",
+	ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380?utm_source=off&utf_medium=web",
```
Comment (teolemon) [inline lib/ProductOpener/Config_obf.pm:182]:
```suggestion
	android_apk_app_link => "https://world.openbeautyfacts.org/images/apps/obf.apk?utm_source=obf&utf_medium=web",
	android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner&utm_source=obf&utf_medium=web",
	ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380?utm_source=obf&utf_medium=web",
```

File: lib/ProductOpener/Config_opff.pm [hunk 4]
```diff
@@ -177,8 +177,8 @@ $flavor = "opff";
 	product_type => "petfood",
 	og_image_url => "https://world.openpetfoodfacts.org/images/misc/openpetfoodfacts-logo-en.png",
 	#android_apk_app_link => "https://world.openbeautyfacts.org/images/apps/obf.apk",
-	#android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner",
-	#ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380",
+	#android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner&utm_source=off&utf_medium=web",
+	#ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380?utm_source=off&utf_medium=web",
```
Comment (teolemon) [inline lib/ProductOpener/Config_opff.pm:181]:
```suggestion
	#android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner&utm_source=opff&utf_medium=web",
	#ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380?utm_source=opff&utf_medium=web",
```

File: lib/ProductOpener/Config_obf.pm [hunk 5]
```diff
@@ -177,9 +177,9 @@ $flavor = 'obf';
 	site_name => "Open Beauty Facts",
 	product_type => "beauty",
 	og_image_url => "https://world.openbeautyfacts.org/images/misc/openbeautyfacts-logo-en.png",
-	android_apk_app_link => "https://world.openbeautyfacts.org/images/apps/obf.apk",
-	android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner",
-	ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380",
+	android_apk_app_link => "https://world.openbeautyfacts.org/images/apps/obf.apk?utm_source=obf&utf_medium=web",
+	android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner&utm_source=obf&utf_medium=web",
+	ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380?utm_source=obf&utf_medium=web",
 	facebook_page_url => "https://www.facebook.com/openbeautyfacts",
```
Comment (teolemon) [inline lib/ProductOpener/Config_obf.pm:183]:
```suggestion
	facebook_page_url => "https://www.facebook.com/openbeautyfacts?utm_source=obf&utf_medium=web",
```

File: lib/ProductOpener/Config_off.pm [hunk 6]
```diff
@@ -196,9 +196,9 @@ $flavor = 'off';
 	site_name => "Open Food Facts",
 	product_type => "food",
 	og_image_url => "https://static.openfoodfacts.org/images/logos/off-logo-vertical-white-social-media-preview.png",
-	android_apk_app_link => "https://world.openfoodfacts.org/files/off.apk",
-	android_app_link => "https://play.google.com/store/apps/details?id=org.openfoodfacts.scanner",
-	ios_app_link => "https://apps.apple.com/app/open-food-facts/id588797948",
+	android_apk_app_link => "https://world.openfoodfacts.org/files/off.apk?utm_source=off&utf_medium=web",
+	android_app_link => "https://play.google.com/store/apps/details?id=org.openfoodfacts.scanner&utm_source=off&utf_medium=web",
+	ios_app_link => "https://apps.apple.com/app/open-food-facts/id588797948?utm_source=off&utf_medium=web",
 	facebook_page_url => "https://www.facebook.com/OpenFoodFacts",
```
Comment (teolemon) [inline lib/ProductOpener/Config_off.pm:202]:
```suggestion
	facebook_page_url => "https://www.facebook.com/OpenFoodFacts?utm_source=off&utf_medium=web",
```

File: lib/ProductOpener/Config_opff.pm [hunk 7]
```diff
@@ -177,8 +177,8 @@ $flavor = "opff";
 	product_type => "petfood",
 	og_image_url => "https://world.openpetfoodfacts.org/images/misc/openpetfoodfacts-logo-en.png",
 	#android_apk_app_link => "https://world.openbeautyfacts.org/images/apps/obf.apk",
-	#android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner",
-	#ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380",
+	#android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner&utm_source=opff&utf_medium=web",
+	#ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380?utm_source=opff&utf_medium=web",
 	#facebook_page_url => "https://www.facebook.com/openbeautyfacts",
```
Comment (teolemon) [inline lib/ProductOpener/Config_opff.pm:182]:
```suggestion
	#facebook_page_url => "https://www.facebook.com/openbeautyfacts?utm_source=opff&utf_medium=web",
```

File: templates/web/common/site_layout.tt.html [hunk 8]
```diff
@@ -242,13 +244,15 @@
 				<!-- full width banner on mobile -->
 				
 				[% IF mobile %]
-					<a href="[% mobile.link %]?utm_source=off&utf_medium=web&utm_campaign=mobile-app-top-banner-[% language %]" id="mobile-app-top-banner" class="button large expand success">
+        <!-- For reference: mobile.link = https://play.google.com/store/apps/details?id=org.openfoodfacts.scanner -->
+					<a href="[% mobile.link %]&utm_source=off&utm_medium=web&utm_campaign=mobile-app-top-banner-[% language %]" id="mobile-app-top-banner" class="button large expand success">
```
Comment (teolemon) [inline templates/web/common/site_layout.tt.html:248]:
```suggestion
					<a href="[% mobile.link %]&utm_campaign=mobile-app-top-banner-[% language %]" id="mobile-app-top-banner" class="button large expand success">
```

File: templates/web/common/site_layout.tt.html (hunk 1 above)
Comment (teolemon) [reply templates/web/common/site_layout.tt.html:55]: get_the_app_link is in po files @stephanegigandet @hangy

File: templates/web/common/site_layout.tt.html [hunk 9]
```diff
@@ -328,8 +332,11 @@ <h1 class="if-empty-dnone">[% title %]</h1>
 							</div>
 						</div>
 						<div class="cell small-100 medium-100 large-50 flex-grid v-align-center direction-row">
+              <!-- msgid "https://apps.apple.com/app/open-beauty-facts/id1122926380" -->
 							<a class="cell small-50 medium-25 large-25 h-space-short v-align-center" href="[% options.ios_app_link %]?utm_source=off&utf_medium=web&utm_campaign=install_the_app_ios_footer_[% language %]"><img src="[% edq(lang('ios_app_icon_url')) %]" alt="[% edq(lang('ios_app_icon_alt_text')) %]"  loading="lazy" class="full-width"></a>
```
Comment (stephanegigandet) [inline templates/web/common/site_layout.tt.html:336]: the utm_source / medium need to be removed for ios too

Comment (teolemon) [inline templates/web/common/site_layout.tt.html:336]:
```suggestion
							<a class="cell small-50 medium-25 large-25 h-space-short v-align-center" href="[% options.ios_app_link %]&utm_campaign=install_the_app_ios_footer_[% language %]"><img src="[% edq(lang('ios_app_icon_url')) %]" alt="[% edq(lang('ios_app_icon_alt_text')) %]"  loading="lazy" class="full-width"></a>
```

**Review by stephanegigandet — APPROVED** [review]
Thank you

File: lib/ProductOpener/Config_obf.pm [hunk 10]
```diff
@@ -177,10 +177,11 @@ $flavor = 'obf';
...
 	og_image_url => "https://world.openbeautyfacts.org/images/misc/openbeautyfacts-logo-en.png",
-	android_apk_app_link => "https://world.openbeautyfacts.org/images/apps/obf.apk",
-	android_app_link => "https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner",
-	ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380",
-	facebook_page_url => "https://www.facebook.com/openbeautyfacts",
+	android_apk_app_link => "https://world.openbeautyfacts.org/images/apps/obf.apk?utm_source=obf&utf_medium=web",
+	android_app_link =>
+		"https://play.google.com/store/apps/details?id=org.openbeautyfacts.scanner&utm_source=obf&utf_medium=web",
+	ios_app_link => "https://apps.apple.com/app/open-beauty-facts/id1122926380?utm_source=obf&utf_medium=web",
+	facebook_page_url => "https://www.facebook.com/openbeautyfacts?utm_source=obf&utf_medium=web",
```
Comment (jayaddison) [inline lib/ProductOpener/Config_obf.pm:184]: @stephanegigandet @teolemon for external domains that we don't manage, adding `utm_source` seems unusual to me -- I don't think that we will receive analytics data about that.  I noticed this during review of #12996 and am thinking of opening a pull request to remove these, to make the URLs cleaner/simpler -- could you let me know?

Comment (teolemon) [reply lib/ProductOpener/Config_obf.pm:184]: in some cases, those services provide some analytics back to us, on facebook, that's unlikely.

Comment (jayaddison) [reply lib/ProductOpener/Config_obf.pm:184]: I found a reference to Google Play providing that in the past -- but was unable to find current documentation.  OK if I open a PR to remove it from non-Google web domains?

**Pull request merged**